salalib/shapemap.cpp

```diff
@@ -1812,14 +1735,14 @@ void ShapeMap::shapeInPolyList(const SalaShape& shape, pvecint& shapeindexlist)
 
 int ShapeMap::testPointInPoly(const Point2f& p, const ShapeRef& shape) const
 {
-   size_t shapeindex = paftl::npos;
+   auto shapeIter = m_shapes.end();
    // simplist: in shape centre
    if (shape.m_tags & ShapeRef::SHAPE_CENTRE) {
-      shapeindex = m_shapes.searchindex(shape.m_shape_ref);
+      shapeIter = m_shapes.find(shape.m_shape_ref);
    }
    // check not an open shape (cannot be inside)
    else if ((shape.m_tags & ShapeRef::SHAPE_OPEN) == 0) {
-      const SalaShape& poly = m_shapes.search(shape.m_shape_ref);
+      const SalaShape& poly = shapeIter->second;

```

Comment:
Erm, wouldn't that dereference `m_shapes.end()`, which would throw if ever invoked?